src/lib.rs

```diff
@@ -40,7 +40,7 @@
 #[macro_use]
 extern crate alloc;
 
-#[cfg(feature = "std")]
+#[cfg(any(feature = "std", feature = "write"))]
```

Comment:
Why not make `write` depend on `std` instead?

Reply:
Because I couldn't figure out how to only enable some dependencies if both `read` and `std` are enabled.